compiler/src/iree/compiler/GlobalOptimization/DemoteContractionInputsToBF16.cpp

```diff
@@ -102,13 +103,17 @@ struct DemoteContractionInputsToBF16Pattern
       replaceOpInputs(static_cast<linalg::BatchMatvecOp *>(nullptr));
     } else if (demoteMatmul && isa<linalg::BatchVecmatOp>(linalgOp)) {
       replaceOpInputs(static_cast<linalg::BatchVecmatOp *>(nullptr));
-    } else if (demoteMatmul && isa<linalg::MatmulTransposeAOp>(linalgOp)) {
+    } else if (demoteMatmul &&
```

Comment:
nit: adding a variable to keep `linalgOp = op.getOperation()` would reduce the number of changes here and other similar places.